File: src/main/java/org/hildan/minecraft/mining/optimizer/display/AnsiCodes.java
package org.hildan.minecraft.mining.optimizer.display;

/**
 * Provides ANSI colorFg codes to print colors in the console.
 */
public class AnsiCodes {

    private static final String RESET = "\u001B[0m";

    private static final String BLACK = "\u001B[30m";

    private static final String RED = "\u001B[31m";

    private static final String GREEN = "\u001B[32m";

    private static final String YELLOW = "\u001B[33m";

    private static final String BLUE = "\u001B[34m";

    private static final String PURPLE = "\u001B[35m";

    private static final String CYAN = "\u001B[36m";

    private static final String WHITE = "\u001B[37m";

    private static final String BLACK_BG = "\u001B[40m";

    private static final String RED_BG = "\u001B[41m";

    private static final String GREEN_BG = "\u001B[42m";

    private static final String YELLOW_BG = "\u001B[43m";

    private static final String BLUE_BG = "\u001B[44m";

    private static final String PURPLE_BG = "\u001B[45m";

    private static final String CYAN_BG = "\u001B[46m";

    private static final String WHITE_BG = "\u001B[47m";

    public static String color(String s, String foregroundColor, String backgroundColor) {
        return foregroundColor + backgroundColor + s + RESET;
    }

    public static String colorFg(String s, String color) {
        return color + s + RESET;
    }

    public static String colorBg(String s, String backgroundColor) {
        return backgroundColor + s + RESET;
    }

    public static String black(String s) {
        return colorBg(s, BLACK);
    }

    public static String red(String s) {
        return colorBg(s, RED);
    }

    public static String green(String s) {
        return colorBg(s, GREEN);
    }

    public static String yellow(String s) {
        return colorBg(s, YELLOW);
    }

    public static String blue(String s) {
        return colorBg(s, BLUE);
    }

    public static String purple(String s) {
        return colorBg(s, PURPLE);
    }

    public static String cyan(String s) {
        return colorBg(s, CYAN);
    }

    public static String white(String s) {
        return colorBg(s, WHITE);
    }
}

File: src/main/kotlin/org/hildan/minecraft/mining/optimizer/display/AnsiCodes.kt
package org.hildan.minecraft.mining.optimizer.display

/**
 * Provides ANSI color codes to print colors in the console.
 */
object AnsiCodes {

    private val RESET = "\u001B[0m"

    val BLACK = "\u001B[30m"

    val RED = "\u001B[31m"

    val GREEN = "\u001B[32m"

    val YELLOW = "\u001B[33m"

    val BLUE = "\u001B[34m"

    val PURPLE = "\u001B[35m"

    val CYAN = "\u001B[36m"

    val WHITE = "\u001B[37m"

    val BLACK_BG = "\u001B[40m"

    val RED_BG = "\u001B[41m"

    val GREEN_BG = "\u001B[42m"

    val YELLOW_BG = "\u001B[43m"

    val BLUE_BG = "\u001B[44m"

    val PURPLE_BG = "\u001B[45m"

    val CYAN_BG = "\u001B[46m"

    val WHITE_BG = "\u001B[47m"

    fun color(s: String, color1: String, color2: String): String {
        return color1 + color2 + s + RESET
    }

    fun color(s: String, color: String): String {
        return color + s + RESET
    }
}
